diff -U3 -r a/src/dosbox.cpp b/src/dosbox.cpp
--- a/src/dosbox.cpp
+++ b/src/dosbox.cpp
@@ -45,12 +45,14 @@
  *      actual leaks among the noise and to patch them up. Thus, "valgrind hunting" --J.C. */
 
 #include <assert.h>
+#include <fstream>
 #include <stdlib.h>
 #include <stdarg.h>
 #include <stdio.h>
 #include <string.h>
 #include <ctime>
 #include <unistd.h>
+#include "SDL_events.h"
 #include "dosbox.h"
 #include "debug.h"
 #include "cpu.h"
@@ -411,6 +413,24 @@
                     return 0;
 #endif
             } else {
+                {
+                    std::ifstream file("input-recording.txt");
+                    while (file.peek() != EOF) {
+                        Bitu t;
+                        char event_type;
+                        int scancode, sym;
+                        file >> t >> event_type >> scancode >> sym;
+                        if (t == PIC_Ticks) {
+                            SDL_Event replay_event = {0};
+                            replay_event.type = event_type == 'u' ? SDL_KEYUP : SDL_KEYDOWN;
+                            replay_event.key.state = event_type == 'u' ? SDL_RELEASED : SDL_PRESSED;
+                            replay_event.key.keysym.scancode = (SDL_Scancode) scancode;
+                            replay_event.key.keysym.sym = sym;
+                            SDL_PushEvent(&replay_event);
+                        }
+                    }
+                    file.close();
+                }
                 GFX_Events();
                 if (DOSBox_Paused() == false && ticksRemain > 0) {
                     TIMER_AddTick();
diff -U3 -r a/src/gui/sdl_mapper.cpp b/src/gui/sdl_mapper.cpp
--- a/src/gui/sdl_mapper.cpp
+++ b/src/gui/sdl_mapper.cpp
@@ -64,6 +64,7 @@
 
 #include <map>
 #include <cctype>
+#include <fstream>
 
 #include <output/output_ttf.h>
 
@@ -4882,10 +4883,18 @@
     }
 
     if (log_keyboard_scan_codes) {
-        if (event->type == SDL_KEYDOWN || event->type == SDL_KEYUP)
+        if (event->type == SDL_KEYDOWN || event->type == SDL_KEYUP) {
             LOG_MSG("MAPPER: SDL keyboard event (%s): scancode=0x%X sym=0x%X mod=0x%X",
                 event->type == SDL_KEYDOWN?"down":"up",
                 event->key.keysym.scancode,event->key.keysym.sym,event->key.keysym.mod);
+            std::ofstream file("input-recording.txt", std::ios_base::app);
+            file << PIC_Ticks
+                 << ' ' << (event->type == SDL_KEYDOWN?"d":"u")
+                 << ' ' << event->key.keysym.scancode
+                 << ' ' << event->key.keysym.sym
+                 << std::endl;
+            file.close();
+        }
     }
 }
 
